dotnet/src/SemanticKernel.Skills/Skills.FileCompression/FileCompressionSkill.cs

```diff
@@ -0,0 +1,158 @@
+    private readonly ILogger<FileCompressionSkill> _logger;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="FileCompressionSkill"/> class.
+    /// </summary>
+    /// <param name="fileCompressor">File compressor implementation</param>
+    /// <param name="logger">Optional logger</param>
+    public FileCompressionSkill(IFileCompressor fileCompressor, ILogger<FileCompressionSkill>? logger = null)
+    {
+        this._fileCompressor = fileCompressor ?? throw new ArgumentNullException(nameof(fileCompressor));
+        this._logger = logger ?? new NullLogger<FileCompressionSkill>();
+    }
+
+    /// <summary>
+    /// Compresses an input file to an output file.
+    /// </summary>
+    /// <param name="sourceFilePath">Path of file to compress</param>
+    /// <param name="context">Semantic Kernel context</param>
+    /// <returns>Path of created compressed file</returns>
+    /// <exception cref="KeyNotFoundException"></exception>
+    [SKFunction("Compresses an input file to an output file")]
+    [SKFunctionInput(Description = "Path of file to compress")]
+    [SKFunctionContextParameter(Name = Parameters.DestinationFilePath, Description = "Path of compressed file to create")]
+    public async Task<string> CompressFileAsync(string sourceFilePath, SKContext context)
+    {
+        this._logger.LogDebug($"{nameof(CompressFileAsync)} got called");
+
+        if (!context.Variables.Get(Parameters.DestinationFilePath, out string destinationFilePath))
+        {
+            const string errorMessage = $"Missing context variable {Parameters.DestinationFilePath} in {nameof(CompressFileAsync)}";
+            this._logger.LogError(errorMessage);
+            context.Fail(errorMessage);
+
+            throw new KeyNotFoundException(errorMessage);
+        }
+
+        await this._fileCompressor.CompressFileAsync(Environment.ExpandEnvironmentVariables(sourceFilePath),
+                                                     Environment.ExpandEnvironmentVariables(destinationFilePath),
+                                                     context.CancellationToken);
+
+        return destinationFilePath;
+    }
+
+    /// <summary>
+    /// Compresses a directory to an output file.
+    /// </summary>
+    /// <param name="sourceDirectoryPath">Path of directory to compress</param>
+    /// <param name="context">Semantic Kernel context</param>
+    /// <returns>Path of created compressed file</returns>
+    /// <exception cref="KeyNotFoundException"></exception>
+    [SKFunction("Compresses a directory to an output file")]
+    [SKFunctionInput(Description = "Path of directory to compress")]
+    [SKFunctionContextParameter(Name = Parameters.DestinationFilePath, Description = "Path of compressed file to create")]
+    public async Task<string> CompressDirectoryAsync(string sourceDirectoryPath, SKContext context)
+    {
+        this._logger.LogDebug($"{nameof(CompressDirectoryAsync)} got called");
```

Comment:
nit: for "got called" kind of logging, I would use `Trace` level

Reply:
Changed